File: CPP/Priority queue/Minimum Operations to Halve Array Sum.cpp
class Solution {
public:
    int halveArray(vector<int>& nums) {
        double sum=accumulate(nums.begin(),nums.end(),0.0);
        priority_queue<double>pq;
        for(auto i:nums){
            pq.push((double)i);
        }
        double half = sum/2.0;
        int total_oper=0;
        double curr_sum=sum;
        while(curr_sum>half){
            double x=pq.top();
            pq.pop();
            double reduced=x/2.0;
            pq.push(reduced);
            curr_sum-=reduced;
            total_oper++;
        }
        return total_oper;
    }
};

File: Python/Priority queue/Minimum Operations to Halve Array Sum.py
class Solution:
    def halveArray(self, nums: List[int]) -> int:
        pq=[]
        sum=0.0
        for num in nums:
            heapq.heappush(pq,-num)
            sum+=num    
        curr_sum=sum
        half=sum/2.0   # // performs integer division, which is incorrect for floating-point calculations.Fix: Use half = sum / 2.0 instead.
        total_oper=0
        while curr_sum>half:
            x=-heappop(pq)
            reduced=x/2.0
            heappush(pq,-reduced)
            curr_sum-=reduced
            total_oper+=1
        return total_oper
